### PR TITLE
Updating samtools

### PR DESCRIPTION
**Pull request by asp8200**
Updating samtools 1.19.2 to 1.20 everywhere - except in two mulled environments (gatk4/markduplicates and minimap2_align).

Got a few pytests which fail on md5 checks. The tests only fail in conda not in docker or singularity. I suggest that we keep the md5 checks, and ignore the failing pytests for conda. (Hopefully such problems will go away when the pytests have been replaced by nf-tests.)

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/gatk4/markduplicates/environment.yml [hunk 1]
```diff
@@ -5,5 +5,5 @@ channels:
   - defaults
 dependencies:
   - bioconda::gatk4=4.5.0.0
-  - bioconda::samtools=1.19.2
```
[the remark this answers is not in the document]
Comment (asp8200) [reply modules/nf-core/gatk4/markduplicates/environment.yml:8]: Right, my bad

[the remark this answers is not in the document]
Comment (asp8200) [reply modules/nf-core/gatk4/markduplicates/environment.yml:8]: Oh, seems like it only a problem in that one `environment.yml`-file, right?

File: modules/nf-core/minimap2/align/environment.yml
Comment (matthdsm) [inline modules/nf-core/minimap2/align/environment.yml:1]: same here

**Review by matthdsm — CHANGES_REQUESTED** [review]
LGTM, save some comments.
Should also really get all the tests to pass

File: modules/nf-core/samtools/calmd/tests/main.nf.test.snap [hunk 2]
```diff
@@ -22,25 +22,25 @@
...
         "content": [
             [
-                "versions.yml:md5,6cdfd8646cf5fb016bd57422bfa00243"
+                "versions.yml:md5,51ce5935fb227e0d66d293442d3a2efd"
             ]
         ],
         "meta": {
             "nf-test": "0.8.4",
-            "nextflow": "24.01.0"
+            "nextflow": "23.10.1"
```
Comment (asp8200) [inline modules/nf-core/samtools/calmd/tests/main.nf.test.snap:42]: @matthdsm : I used nf 23.10.1 when updating the test.snaps. Would it have been better to use 24.01.0 as that was what was previously used for these tests?

Comment (matthdsm) [reply modules/nf-core/samtools/calmd/tests/main.nf.test.snap:42]: I think it should match the nf version that's used in to CI?

**Comment by asp8200** [comment]
> LGTM, save some comments. Should also really get all the tests to pass

Thanks for the review and comments. Do you have any suggestion on how to get the failing conda-based pytests passing? The only way, I can think of, is to remove the md5 checks in those pytests, but that will also remove them in the docker- and singularity-based pytests where md5 checks are not failing.  🤔

**Comment by matthdsm** [comment]
> > LGTM, save some comments. Should also really get all the tests to pass
> 
> Thanks for the review and comments. Do you have any suggestion on how to get the failing conda-based pytests passing? The only way, I can think of, is to remove the md5 checks in those pytests, but that will also remove them in the docker- and singularity-based pytests where md5 checks are not failing. 🤔

Migrating everything to pytest is the way to go right now I think...

File: modules/nf-core/samtools/calmd/tests/main.nf.test.snap (hunk 2 above)
Comment (asp8200) [reply modules/nf-core/samtools/calmd/tests/main.nf.test.snap:42]: Oh, seems like the CI tests are using the latest NF, that is, v24.4.2 which was released two hours ago :-) I'll try to see if I can update the test.snaps with NF v24.4.2.

**Comment by maxulysse** [comment]
> > > LGTM, save some comments. Should also really get all the tests to pass
> > 
> > 
> > Thanks for the review and comments. Do you have any suggestion on how to get the failing conda-based pytests passing? The only way, I can think of, is to remove the md5 checks in those pytests, but that will also remove them in the docker- and singularity-based pytests where md5 checks are not failing. 🤔
> 
> Migrating everything to nf-test is the way to go right now I think...

I'm willing not to put the migration burden on @asp8200 who is nice enough to try and upgrade everything

**Comment by fellen31** [comment]
I can update minimap2_align.

**Review by nvnieuwk — APPROVED** [review]
I did not look at this in detail, but seeing those tests pass makes it LGTM right now :)

**Pull request merged**